core/src/main/java/jenkins/tasks/SimpleBuildWrapper.java

```diff
@@ -143,6 +268,7 @@ protected boolean runPreCheckout() {
             return new Environment() {};
         } else {
             final Context c = new Context();
+            c.setWorkspaceRequirement(this);
```

Comment:
Purely stylistic, but to make all this logic easier to follow:

* remove `setWorkspaceRequirement`
* deprecate the implicit no-arg `Context` constructor (by first declaring it explicitly)
* introduce a private (or package-scope) constructor taking a `boolean requiresWorkspace` argument
* call that from a new “beta” method `SimpleBuildWrapper.createContext()`
* then you can

```suggestion
            final Context c = createContext();
```

Reply:
OK

Reply:
Done